code/API_definitions/Application_Endpoint_Discovery_API.yaml

```diff
@@ -0,0 +1,596 @@
+          required: false
+          description: |
+            When the API Consumer includes the "X-Correlator" header in the
+            request, the API provider must include it in the response with
+            the same value that was used in the request. Otherwise, it is
+            optional to include the "X-Correlator" header in the response with
+             any valid value. Recommendation is to use UUID for values.
+          schema:
+            type: string
+            format: uuid
+      responses:
+        "200":
+          description: |
+            Successful response, returning the closest App Instance Endpoint
+            Zone to the user device identified in the request header.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/AccessEndpoint"
+        "400":
+          $ref: "#/components/responses/400BadRequest"
+        "401":
+          $ref: "#/components/responses/401Unauthorized"
+        "403":
+          $ref: "#/components/responses/403Forbidden"
+        "404":
+          $ref: "#/components/responses/404NotFound"
+        "405":
+          $ref: "#/components/responses/405MethodNotAllowed"
+        "406":
+          $ref: "#/components/responses/406Unacceptable"
+        "429":
+          $ref: "#/components/responses/429TooManyRequests"
+        "500":
+          $ref: "#/components/responses/500InternalServerError"
+        "502":
+          $ref: "#/components/responses/502BadGateway"
+        "503":
+          $ref: "#/components/responses/503ServiceUnavailable"
+        "504":
+          $ref: "#/components/responses/504GatewayTimeout"
+      tags:
+        - Discovery
+      summary: |
+        Returns the Endpoint of the closest App Instance to user device
+        identified in the request.
+      description: |
+        On receiving this request, the network will return the name
+        of the App Instance with the shortest network path to the end user
+        device identified in the request.
+components:
+  securitySchemes:
+    openId:
+      description: OpenID Provider Configuration Information.
+      type: openIdConnect
+      openIdConnectUrl: .well-known/openid-configuration
+  headers:
+    X-Correlator:
+      description: |
+       When the API Consumer includes the "X-Correlator" header in the request,
+        the API provider must include it in the response with the same value t
+        hat was used in the request. Otherwise, it is optional to include the
+        "X-Correlator" header in the response with any valid value.
+        Recommendation is to use UUID for values.
+      required: false
+      schema:
+        type: string
+        format: uuid
+  schemas:
+    AppId:
+      type: string
+      format: uuid
+      description: A globally unique identifier associated with the application.
+        OP generates this identifier when the application is submitted over NBI.
+    AccessEndpoint:
```

Comment:
As commented earlier an application may have more than one endpoints and for that reason we may use AccessEndpoints to accommodate plurality. If so, then we would need to make the AccessEndpoints as an array of objects that is being defined below which contains the endpoint accessibility information?

Reply:
Also, in my opinion if multiple endpoints are returned then there should be an additional field which allow to embed a purpose or qualifier for a given endpoint which can be used by the client applications to differentiate one endpoint with other e.g. low quality video stream or high quality video stream endpoint? But then these qualifier should come from the API invoker may be at application onboarding time. Any thoughts?